### ISSUE TITLE
Cannot add games to steam

### ISSUE THREAD

**Issue by Rinma** [Heroic-Games-Launcher/HeroicGamesLauncher#3647]
### Describe the bug

When I try to add a game from Epic to steam I get the error `Cannot convert undefined or null to object`.
I use a Steam Deck with the Flatpak version from Heroic Games Launcher in Version 2.13.0

### Add logs

```text
Error occurred in handler for 'isAddedToSteam': TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
    at getAppName (/app/bin/heroic/resources/app.asar/build/electron/main.ac42ed07.js:8:1281)
    at /app/bin/heroic/resources/app.asar/build/electron/main.ac42ed07.js:8:1856
    at Array.findIndex (<anonymous>)
    at checkIfAlreadyAdded (/app/bin/heroic/resources/app.asar/build/electron/main.ac42ed07.js:8:1843)
    at isAddedToSteam (/app/bin/heroic/resources/app.asar/build/electron/main.ac42ed07.js:11:684)
```

```
Error occurred in handler for 'addToSteam': TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
    at getAppName (/app/bin/heroic/resources/app.asar/build/electron/main.ac42ed07.js:8:1281)
    at /app/bin/heroic/resources/app.asar/build/electron/main.ac42ed07.js:8:1856
    at Array.findIndex (<anonymous>)
    at checkIfAlreadyAdded (/app/bin/heroic/resources/app.asar/build/electron/main.ac42ed07.js:8:1843)
    at addNonSteamGame (/app/bin/heroic/resources/app.asar/build/electron/main.ac42ed07.js:8:2642)
    at async WebContents.<anonymous> (node:electron/js2c/browser_init:2:88992)
```

### Steps to reproduce

1. Start the Flatpak version from HGL from the terminal `flatpak run com.heroicgameslauncher.hgl`
2. Install game from Epic
3. Go to game page and click "add to steam"
![image](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/1405824/26dfea2d-b0a2-4371-b691-72ed15c76c22)
4. See the error in the terminal

### Expected behavior

I can add games to steam library

### Heroic Version

Latest Stable

### System Information

Operating System: SteamOS 3.5.17
KDE Plasma Version: 5.27.5
KDE Frameworks Version: 5.107.0
Qt Version: 5.15.9
Kernel Version: 6.1.52-valve16-1-neptune-61 (64-bit)
Graphics Platform: X11
Processors: 8 × AMD Custom APU 0405
Memory: 14.5 GiB of RAM
Graphics Processor: AMD Custom GPU 0405
Manufacturer: Valve
Product Name: Jupiter
System Version: 1

**Comment by codethought** [comment]
I've seen similar behavior here (but have not looked in the logs to see if there is an error message). Restarting steam enables the visibility of the game I attempted to "Add to Steam" from the launcher.

**Comment by arielj** [comment]
> I've seen similar behavior here (but have not looked in the logs to see if there is an error message). Restarting steam enables the visibility of the game I attempted to "Add to Steam" from the launcher.

this is normal, after adding a game to steam you have to restart steam for it to reload the list of non-steam games, the notification message tells you that

this report if for the games not added at all (not even after a restart of steam)

**Comment by codethought** [comment]
>this report if for the games not added at all (not even after a restart of steam)

Got it.. thank you.

**Comment by codethought** [comment]
> this is normal, after adding a game to steam you have to restart steam for it to reload the list of non-steam games, the notification message tells you that

Actually I went and ran again.. there was no notification to that effect.. that's why I got confused.. :)

**Comment by Rinma** [comment]
Seems to be fixed with Update 2.14.1
